submissions/misshatsker/object-oriented-js/app.js

```diff
@@ -0,0 +1,188 @@
+const leftSide = 0;
+const topSide = 0;
+const rightSide = 500;
+const bottomSide = 500;
+
+const initialGroundY = 60;
+const tileHeight = 80;
+
+const enemySprite = 'images/enemy-bug.png';
+const enemyWidth = 80;
+const enemyHeight = 70;
+const enemyMinSpeed = 100;
+const enemyMaxSpeed = 300;
+const enemyMinX = -rightSide;
+const enemyMaxX = enemyWidth - rightSide;
+
+const playerSprite = 'images/char-princess-girl.png';
+const playerWidth = 70;
+const playerHeight = 70;
+const playerSpeed = 75;
+const playerInitialX = 200;
+const playerInitialY = 400;
+
+const Character = function(x, y, speed, sprite) {
+    this.sprite = sprite;
+    this.speed = speed;
+    this.x = x;
+    this.y = y;
+};
+
+Character.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const Enemy = function(...args) {
+    Character.call(this, ...args);
+};
+
+Enemy.prototype.update = function(dt) {
+    const shift = this.speed * dt;
+    this.x = this.x + shift;
+    if (this.x >= rightSide) {
+        this.x = -this.width;
+    }
+};
+
+Enemy.prototype.width = enemyWidth;
+Enemy.prototype.height = enemyHeight;
```

Comment:
Why mixed style of assigning class properties?

Reply:
You're right, they shouldn't be there. I considered them as common properties, but they aren't